app/models/order_detail.rb

```diff
@@ -771,6 +747,34 @@ def problem_order?
     problem
   end
 
+  def can_dispute?
+    in_review? && dispute_at.blank?
+  end
+
+  def in_dispute?
+    dispute_at.present? && dispute_resolved_at.nil? && !canceled?
+  end
+
+  def in_review?
+    reviewed_at.try(:future?) && !in_dispute?
+  end
+
+  def reviewed?
+    reviewed_at.try(:past?) && !in_dispute?
+  end
```

Comment:
Trying to understand this... and the `past?` bit means we don't also need to check `in_review?`, right?

It's also just weird that `present?` and `past?` mean totally different things.